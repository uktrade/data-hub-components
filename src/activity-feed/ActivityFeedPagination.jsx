import React from 'react'
import Button from '@govuk-react/button'
import LoadingBox from '@govuk-react/loading-box'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { SPACING } from '@govuk-react/constants'

const Pagination = styled('div')`
  text-align: center;
  margin-top: ${SPACING.SCALE_5};
  margin-bottom: ${SPACING.SCALE_6};
`

export default class ActivityFeedPagination extends React.Component {
  static propTypes = {
    onLoadMore: PropTypes.func,
    isLoading: PropTypes.bool,
  }

  static defaultProps = {
    onLoadMore: () => {},
    isLoading: false,
  }

  render() {
    const { onLoadMore, isLoading } = this.props

    return (
      <Pagination>
        <LoadingBox loading={isLoading} backgroundColorOpacity={1} timeOut={0}>
          <Button
            disabled={isLoading}
            onClick={onLoadMore}
            buttonColour="#dee0e2"
            buttonTextColour="#000"
          >
            Show more activity
          </Button>
        </LoadingBox>
      </Pagination>
    )
  }
}
