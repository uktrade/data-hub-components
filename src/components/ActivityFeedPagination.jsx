import React from 'react'

export default class ActivityFeedPagination extends React.Component {
	render() {
    return (
      <button>Show more activity</button>
    )
  }
}
