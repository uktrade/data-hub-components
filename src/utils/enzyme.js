export const withTargetValue = value => ({
  target: {
    value,
  },
})

/**
 * @function getOptionValueByText
 * @description Finds an `<option>` element with its inner text matching the
 * {text} parameter and returns the value of its `value` attribute.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const getOptionValueByText = (wrapper, text) => (
  wrapper.find('option')
    .find({ children: text })
    .props()
    .value
)

/**
 * @function changeAndUpdate
 * @description Simulates the `change` event on the provided {wrapper} with
 * {vaule} as `target.value` and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {*} value - The value to be used as `event.target`
 */
export const changeAndUpdate = (wrapper, value) => {
  wrapper.simulate('change', withTargetValue(value))
  wrapper.update()
}

/**
 * @function changeSelectAndUpdate
 * @description Finds a `<select>` element in the wrapper, selects the option
 * matching the {text} and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const changeSelectAndUpdate = (wrapper, text) => {
  changeAndUpdate(wrapper.find('select'), getOptionValueByText(wrapper, text))
  wrapper.update()
}
