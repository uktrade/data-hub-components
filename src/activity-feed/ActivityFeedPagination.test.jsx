import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import ActivityFeedPagination from './ActivityFeedPagination'

describe('ActivityFeedPagination', () => {
  test('renders default pagination', () => {
    const tree = renderer
      .create(<ActivityFeedPagination />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders pagination with loader', () => {
    const tree = renderer
      .create(
        <ActivityFeedPagination isLoading={true} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders pagination with loader', () => {
    const tree = renderer
      .create(
        <ActivityFeedPagination
          isLoading={true}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('simulates click events', () => {
    const onLoadMoreClick = jest.fn()
    const wrapper = <ActivityFeedPagination
      isLoading={false}
      onLoadMore={onLoadMoreClick}
    />

    mount(wrapper)
      .find('button')
      .simulate('click')
    expect(onLoadMoreClick).toHaveBeenCalledTimes(1)
  })
})




