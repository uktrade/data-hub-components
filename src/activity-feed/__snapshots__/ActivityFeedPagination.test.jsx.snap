// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFeedPagination renders default pagination 1`] = `
<div
  className="sc-cmthru dhFBXA"
>
  <div
    className="sc-eqIVtm fxHPMe"
  >
    <div
      className="sc-fAjcbJ fsjQAG"
    >
      <svg
        className="icon-loading"
        fill="#0b0c0c"
        height="50px"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "display": "block",
          }
        }
        version="1.1"
        viewBox="-25 -25 50 50"
        width="50px"
      >
        <title>
          Loading
        </title>
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "0ms",
            }
          }
          transform="rotate(0, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "83ms",
            }
          }
          transform="rotate(30, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "166ms",
            }
          }
          transform="rotate(60, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "249ms",
            }
          }
          transform="rotate(90, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "332ms",
            }
          }
          transform="rotate(120, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "415ms",
            }
          }
          transform="rotate(150, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "498ms",
            }
          }
          transform="rotate(180, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "581ms",
            }
          }
          transform="rotate(210, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "664ms",
            }
          }
          transform="rotate(240, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "747ms",
            }
          }
          transform="rotate(270, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "830ms",
            }
          }
          transform="rotate(300, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "913ms",
            }
          }
          transform="rotate(330, 0, 2) translate(10 0)"
          width="12"
        />
      </svg>
      <div
        className="overlay sc-caSCKo cQyEyL"
      />
    </div>
    <button
      className="sc-VigVT kJjasd"
      disabled={false}
      onClick={[Function]}
    >
      Show more activity
    </button>
  </div>
</div>
`;

exports[`ActivityFeedPagination renders pagination with loader 1`] = `
<div
  className="sc-cmthru dhFBXA"
>
  <div
    className="sc-eqIVtm fxHPMe"
  >
    <div
      className="sc-fAjcbJ fsjQAG"
    >
      <svg
        className="icon-loading"
        fill="#0b0c0c"
        height="50px"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "display": "block",
          }
        }
        version="1.1"
        viewBox="-25 -25 50 50"
        width="50px"
      >
        <title>
          Loading
        </title>
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "0ms",
            }
          }
          transform="rotate(0, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "83ms",
            }
          }
          transform="rotate(30, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "166ms",
            }
          }
          transform="rotate(60, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "249ms",
            }
          }
          transform="rotate(90, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "332ms",
            }
          }
          transform="rotate(120, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "415ms",
            }
          }
          transform="rotate(150, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "498ms",
            }
          }
          transform="rotate(180, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "581ms",
            }
          }
          transform="rotate(210, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "664ms",
            }
          }
          transform="rotate(240, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "747ms",
            }
          }
          transform="rotate(270, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "830ms",
            }
          }
          transform="rotate(300, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "913ms",
            }
          }
          transform="rotate(330, 0, 2) translate(10 0)"
          width="12"
        />
      </svg>
      <div
        className="overlay sc-caSCKo cQyEyL"
      />
    </div>
    <button
      className="sc-VigVT kJjasd"
      disabled={true}
      onClick={[Function]}
    >
      Show more activity
    </button>
  </div>
</div>
`;

exports[`ActivityFeedPagination renders pagination with loader 2`] = `
<div
  className="sc-cmthru dhFBXA"
>
  <div
    className="sc-eqIVtm fxHPMe"
  >
    <div
      className="sc-fAjcbJ fsjQAG"
    >
      <svg
        className="icon-loading"
        fill="#0b0c0c"
        height="50px"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "display": "block",
          }
        }
        version="1.1"
        viewBox="-25 -25 50 50"
        width="50px"
      >
        <title>
          Loading
        </title>
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "0ms",
            }
          }
          transform="rotate(0, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "83ms",
            }
          }
          transform="rotate(30, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "166ms",
            }
          }
          transform="rotate(60, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "249ms",
            }
          }
          transform="rotate(90, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "332ms",
            }
          }
          transform="rotate(120, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "415ms",
            }
          }
          transform="rotate(150, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "498ms",
            }
          }
          transform="rotate(180, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "581ms",
            }
          }
          transform="rotate(210, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "664ms",
            }
          }
          transform="rotate(240, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "747ms",
            }
          }
          transform="rotate(270, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "830ms",
            }
          }
          transform="rotate(300, 0, 2) translate(10 0)"
          width="12"
        />
        <rect
          className="Spinner__Rect-sc-1pm4zwa-0 glkcMw"
          fill="#0b0c0c"
          height="5"
          opacity="0"
          rx="2.5"
          ry="2.5"
          style={
            Object {
              "animationDelay": "913ms",
            }
          }
          transform="rotate(330, 0, 2) translate(10 0)"
          width="12"
        />
      </svg>
      <div
        className="overlay sc-caSCKo cQyEyL"
      />
    </div>
    <button
      className="sc-VigVT kJjasd"
      disabled={true}
      onClick={[Function]}
    >
      Show more activity
    </button>
  </div>
</div>
`;
